### DISCUSSION TITLE
USB to ethernet adapters not showing up as interfaces

### DISCUSSION THREAD

**Discussion by antoinefaure** [OE4T/meta-tegra#1671] [Q&A] [answered]
Hi everyone,

I'm using a [Seeedstudio reComputer J4012](https://www.seeedstudio.com/reComputer-J4012-p-5586.html) (`p3768-0000-p3767-0000` machine)

After updating from the `kirkstone` branch to `scarthgap`, my USB to ethernet adapters are detected by `lsusb` but don't end up as an actual network interface. 

This was working fine when using the `kirkstone` branch. My project also includes `meta-raspberrypi`, using the same distro and same image. Even when using the `scarthgap` branch the network interfaces are brought up properly on the raspberry pi. 

The tegra's kernel seems to include everything that I would need:
```
CONFIG_USB_NET_DRIVERS=y
CONFIG_USB_PEGASUS=m
CONFIG_USB_RTL8150=m
CONFIG_USB_RTL8152=m
CONFIG_USB_LAN78XX=m
CONFIG_USB_USBNET=m
CONFIG_USB_NET_AX8817X=m
CONFIG_USB_NET_AX88179_178A=m
```

The [dongle](https://www.level1.com/level1_en/usb-0301-fast-ethernet-usb-network-adapter-54002307601) I am using is based on an RTL8152 which is included in the config, and is detected:
```
lsusb
[...]
Bus 001 Device 003: ID 0bda:8152 Realtek Semiconductor Corp. RTL8152 Fast Ethernet Adapter
```

And when connecting the dongle I only get the following log:
`[  106.504551] usb 1-2.3: new high-speed USB device number 4 using tegra-xusb`

I have the same dependencies on `usbmuxd`,  `usb-modeswitch` and `usb-modeswitch-data`  and don't see any other difference. 

If someone has any hints or ideas, any help would be appreciated!

Thanks in advance,
Antoine

**Comment by joekale** [comment, marked as answer]
You may need to install `kernel-module-r8152` to your image to ensure the module is included in your image.

**Reply by antoinefaure** [reply]
Very good point! I forgot that the kernel modules were packaged separately. Either adding `kernel-module-r8152` to the image or the more generic `IMAGE_INSTALL += "kernel-modules"`  fixed it.
Thanks a lot @joekale !
